sycl/include/sycl/ext/intel/esimd/memory.hpp

```diff
@@ -2593,52 +2631,406 @@ gather_impl(AccessorTy acc, simd<uint32_t, N> offsets, uint32_t glob_offset,
+///                   simd_mask<N> mask, PropertyListT props       // (acc-ga-5)
+/// This function is identical to (acc-ga-2) except that vector size is fixed
+/// to 1. This variant is added for convenience and let user omit the template
+/// arguments and call the function as 'gather(acc, byte_offsets, mask);'.
+template <typename T, int N, typename AccessorT, typename OffsetT,
+          typename MaskT,
+          typename PropertyListT =
+              ext::oneapi::experimental::detail::empty_properties_t>
+__ESIMD_API std::enable_if_t<
+    (detail::is_device_accessor_with_v<AccessorT,
+                                       detail::accessor_mode_cap::can_read> &&
+     ext::oneapi::experimental::is_property_list_v<PropertyListT> &&
+     std::is_same_v<MaskT, simd_mask<N>>),
+    simd<T, N>>
+gather(AccessorT acc, simd<OffsetT, N> byte_offsets, MaskT mask,
+       PropertyListT props = {}) {
+  return gather<T, N, 1>(acc, byte_offsets, mask, props);
+}
+
+/// template <typename T, int N, typename AccessorT, typename OffsetT,
+///           typename PropertyListT = empty_properties_t>
+/// simd<T, N> gather(AccessorT acc, simd<OffsetT, N> byte_offsets,
+///                   PropertyListT props = {});                   // (acc-ga-6)
+/// This function is identical to (acc-ga-3) except that vector size is fixed
+/// to 1. This variant is added for convenience and let user omit the template
+/// arguments and call the function as 'gather(acc, byte_offsets);'.
+template <typename T, int N, typename AccessorT, typename OffsetT,
+          typename PropertyListT =
+              ext::oneapi::experimental::detail::empty_properties_t>
+__ESIMD_API std::enable_if_t<
+    (detail::is_device_accessor_with_v<AccessorT,
+                                       detail::accessor_mode_cap::can_read> &&
+     ext::oneapi::experimental::is_property_list_v<PropertyListT>),
+    simd<T, N>>
+gather(AccessorT acc, simd<OffsetT, N> byte_offsets, PropertyListT props = {}) {
+  return gather<T, N, 1>(acc, byte_offsets, props);
+}
+
+/// template <typename T, int N, int VS = 1, typename AccessorT,
+///           typename OffsetSimdViewT,
+//            typename PropertyListT = empty_properties_t>
+/// simd<T, N> gather(AccessorT acc, simd<OffsetT, N / VS> byte_offsets,
+///                   simd_mask<N / VS> mask, simd<T, N> pass_thru,
+///                   PropertyListT props = {});                   // (acc-ga-7)
+/// This function is identical to (acc-ga-1) except that the \p byte_offsets
+/// is represented as \c simd_view.
+template <typename T, int N, int VS = 1, typename AccessorT,
+          typename OffsetSimdViewT,
+          typename PropertyListT =
+              ext::oneapi::experimental::detail::empty_properties_t>
+__ESIMD_API std::enable_if_t<
+    (detail::is_device_accessor_with_v<AccessorT,
+                                       detail::accessor_mode_cap::can_read> &&
+     detail::is_simd_view_type_v<OffsetSimdViewT> &&
```

Comment:
I don't think we did simd_view SFINAE checks in the other APIs, we may want to go back and add those. or do we only need it here because of the ambiguous call problem?

Reply:
IMO, it was more convenient way to do the same.
For ambiguous call problem I added `MaskT` template arg to (acc-ga-5), which conflicted with older function `gather(acc, simd byte_offsets, OffsetT global_offset` in such calls: `auto res = gather<T, N>(acc, byte_offsets, 0);`
`0` could be read as `OffsetT global_offset` from the old prototype, or `mask<N>` from (acc-ga-5)`.